### ISSUE TITLE
Build and publish the Dart language specification using cloud build

### ISSUE THREAD

**Issue by whesse** [dart-lang/sdk#52903]
The Dart language specification is currently built and published using GitHub actions. This requires giving the github action a token to publish the document to a Google cloud web site.

Change these steps to run on Google cloud build instead, which will automatically run them with credentials to publish to this web site.

@whesse @eernstg

**Comment by eernstg** [comment]
Sounds very good! Do I need to make any changes to the language repository to make that work?

**Comment by whesse** [comment]
Once the cloud build is set up, and triggered by changes to the repository, then the existing github action can be removed. I don't think any other changes are needed. We could put the steps taken by the cloud builder into a script that is in the language repository. That way, any changes to how the build builds and uploads the spec could be made just by changing that script. But that also makes it a little less secure.

**Comment by mit-mit** [comment]
I already made a bot account for this: https://github.com/dart-spec-bot

The credentials are stored internally under `dart-spec-bot`

**Comment by whesse** [comment]
I'm not sure what the GitHub dart-spec-bot account would be used for.
I think that a cloud builder, set up to trigger when a commit to dart-lang/language that changes the spec lands, would do the creation of the latex and html, and upload it to the Firebase hosting, all using a service account in a GCP project.

https://cloud.google.com/build/docs/triggers

**Comment by whesse** [comment]
You are right, the connection from GitHub to cloud build needs to be configured by a GitHub user account that has admin or owner rights. I will use the dart-spec-bot account to configure the connection. It looks like this connection may  "authorize the Cloud Build GitHub App to access your GitHub account", so it would be using this account.

**Comment by devoncarew** [comment]
FYI, I believe that @khanhnwin has done similar work wrt cloud builders.

**Comment by eernstg** [comment]
:tada:
